### PR TITLE
API Deletes trigger archive instead of delete

### PR DESCRIPTION
**Pull request by tractorcow**
Merge with https://github.com/silverstripe/silverstripe-framework/pull/6195

### CODE REVIEW

File: code/Controller/AssetAdmin.php [hunk 1]
```diff
@@ -288,15 +288,15 @@ public function apiDelete(HTTPRequest $request)
         }
 
         if (!min(array_map(function (File $file) {
-            return $file->canDelete();
+            return $file->canArchive();
```
Comment (flamerohr) [inline code/Controller/AssetAdmin.php]: Is this necessary? Seems like it's only for testing purposes and many other places still use `canDelete()`

Comment (sminnee) [inline code/Controller/AssetAdmin.php]: Only the test implementation is for testing purposes. `Versioned::canArchive()` provides the real version.

File: tests/php/AssetAdminTest.php [hunk 2]
```diff
@@ -502,6 +502,14 @@ public function canDelete($member = null)
         }
     }
 
+    public function canArchive($member = null)
+    {
+        if ($this->owner->Name === 'disallowCanDelete.txt') {
+            return false;
+        }
+        return $this->owner->canDelete($member);
```
Comment (sminnee) [inline tests/php/AssetAdminTest.php]: Shouldn't this be calling the Versioned::canArchive() code, or returning `null` to result in a no-op?

File: code/Controller/AssetAdmin.php (hunk 1 above)
Comment (flamerohr) [inline code/Controller/AssetAdmin.php]: Sorry, I jumped that assumption about testing :D
What I mean is shouldn't this switch to `canArchive()` carry through to the `canDelete()` calls in functions such as `getObjectFromData()`?
It seemed weird to have `canArchive()` in the `apiDelete()` function but the schemas still reflect `canDelete()`

File: tests/php/AssetAdminTest.php (hunk 2 above)
Comment (tractorcow) [inline tests/php/AssetAdminTest.php]: This is a work around to an annoying issue with multiple extensions both applying a default method implementation.

a canArchive here would be an infinity loop.

File: code/Controller/AssetAdmin.php (hunk 1 above)
Comment (tractorcow) [inline code/Controller/AssetAdmin.php]: Yes I should apply it there too.

**Pull request merged**